### PR TITLE
test(test-plugin-ledger-connector-besu): jestify get-block-endpoint

### PR DESCRIPTION
**Pull request by krriscode**
Migration of get-block-endpoint.ts from tape-promise to Jest.
Update tests for improved reliability and performance.

```
packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/
integration/plugin-validator-besu/get-block-endpoint.test.ts
```

Fixes: #3500

Co-authored-by: Peter Somogyvari <peter.somogyvari@accenture.com>

Signed-off-by: krriscode <85664362+krriscode@users.noreply.github.com>
Signed-off-by: Peter Somogyvari <peter.somogyvari@accenture.com>

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Review by petermetz — APPROVED** [review]
@krriscode Looking good, thank you!

**Review by petermetz — COMMENTED** [review]
@krriscode Please fix the commit lint check by modifying the git commit message. It looks like you need a line break somewhere to make the lines less than 100 characters each.

If you need any help with the git mechanics involved just let me know and I can help.

image

**Review by petermetz — CHANGES_REQUESTED** [review]
@krriscode Getting closer! Now the commit message is looking good but you are not doing the signoff with the correct name and/or email address. Please see the details here:

https://github.com/hyperledger-cacti/cacti/pull/3607/checks?check_run_id=32664817913

**Review by petermetz — CHANGES_REQUESTED** [review]
@krriscode Same issue still. There's a more detailed explanation of how it works that might help if you read all of that: https://stackoverflow.com/a/62972983/698470

File: .taprc [hunk 1]
```diff
@@ -24,8 +24,19 @@ files:
   - ./packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/v21-get-block-endpoint.test.ts
   - ./packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-past-logs-endpoint.test.ts
   - ./packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-transaction-endpoint.test.ts
+<<<<<<< HEAD
```
Comment (petermetz) [inline .taprc:27]: @krriscode I just noticed these, please do a conflict resolution here so that the diff only includes your changes.

**Review by petermetz — CHANGES_REQUESTED** [review]
@krriscode The sign off is still not working. You are trying to submit your pull request being signed off as someone else:

```sh
 Error: The SoB (DCO) check failed

  9f62e82005578755f233384042f0b0068e5fea6a    Expected "Udhayakumari <62878965+Udhayakumari@users.noreply.github.com>", but got "krriscode <85664362+krriscode@users.noreply.github.com>".

  What should I do to fix it ?

  All proposed commits should include a Signed-off-by: <your-name> <your-email-address> line in their commit message.
  This is most conveniently done by using --signoff (-s) when running git commit.
```

Source: https://github.com/hyperledger-cacti/cacti/actions/runs/11819193500/job/32928793439

**Comment by krriscode** [comment]
Hi @petermetz, 
I have check git config twice and it is correct, below is screenshot of it
I got stuck here could you help here how to resolve something issue.
image

File: .taprc (hunk 1 above)
Comment (krriscode) [reply .taprc]: @petermetz, I have resolved the conflict of .taprc file, please re-review it.

Comment (petermetz) [reply .taprc]: @krriscode Thank you! Looks good!

**Review by petermetz — APPROVED** [review]
@krriscode LGTM, thank you!

**Comment by petermetz** [comment]
@krriscode Upon further investigation it looks like you did manage to fix it after all, so we are good to go now we just need other maintainers to review it and then it can be merged. Thank you again for the contribution!

**Review by outSH — APPROVED** [review]

**Review by petermetz — CHANGES_REQUESTED** [review]
@krriscode I'm not sure what happened but somewhere along the way the diff added a bunch of unrelated files to `.taprc` and `jest.config.js`.

Please ensure that the diff only makes changes to the relevant lines (the line of code with the file path of the test case being migrated)

image

**Comment by petermetz** [comment]
@krriscode I'll try to resolve the issues myself to help out. FYI: I'll do a force push to your branch

**Review by petermetz — APPROVED** [review]
@krriscode Should be good to go now. LGTM

**Comment by petermetz** [comment]
I'm going to fix the linter errors in a minute

**Comment by petermetz** [comment]
@krriscode All is good now, about to merge it. Thank you again for the contribution and welcome to the Cacti community! We hope that you'll be back for more! :-)

**Comment by krriscode** [comment]
Hi @petermetz,
I truly appreciate your help and guidance with the PR. Your efforts in resolving the issues and ensuring this contribution was successful mean a lot to me. Thank you so much!

**Pull request merged**